library/core/src/intrinsics.rs

```diff
@@ -2291,6 +2291,16 @@ extern "rust-intrinsic" {
     /// [`std::hint::black_box`]: crate::hint::black_box
     #[rustc_const_unstable(feature = "const_black_box", issue = "none")]
     pub fn black_box<T>(dummy: T) -> T;
+
+    /// `ptr` must point to a vtable.
+    /// The intrinsic will return the size stored in that vtable.
+    #[cfg(not(bootstrap))]
+    pub fn vtable_size(ptr: *const ()) -> usize;
+
+    /// `ptr` must point to a vtable.
+    /// The intrinsic will return the alignment stored in that vtable.
+    #[cfg(not(bootstrap))]
+    pub fn vtable_align(ptr: *const ()) -> usize;
```

Comment:
We may be able to avoid these by creating a dummy `(*const (), &'static Vtable)`, transmuting it to `&dyn Send` and reusing the existing `size_of_val`.

Reply:
The other option is to implement the existing `size_of_val` and `align_of_val` in terms of these intrinsics. I don't know if MIR optimizations are good enough already to do that efficiently even in debug mode though.

Reply:
I considered that but then realized that these are *not the same* as size_of_val/align_of_val. I should have left a comment.

In `&(i32, dyn Send)`, the vtable will contain the size of the `dyn` part, but size_of_val will return that size + 4! (And maybe more than that due to alignment.)

Reply:
oof right.